### PR TITLE
Add SbPlayer telemetry

### PR DESCRIPTION
**Pull request by sideb0ard**
Adding additional Media Telemetry to cover SbPlayer create and destruction.

b/303328238

### CODE REVIEW

**Review by at-ninja — APPROVED** [review]

File: cobalt/media/base/metrics_provider.cc [hunk 1]
```diff
@@ -121,21 +121,38 @@ void MediaMetricsProvider::EndTrackingAction(WebMediaPlayerAction action) {
...
+      UMA_HISTOGRAM_CUSTOM_MICROSECONDS_TIMES(
+          "Cobalt.Media.SbPlayer.CreateUrlPlayer.Timing", action_duration,
+          base::TimeDelta::FromMicroseconds(100),
+          base::TimeDelta::FromMilliseconds(5000), 50);
+      break;
+    case MediaAction::SBPLAYER_DESTROY:
+      UMA_HISTOGRAM_CUSTOM_MICROSECONDS_TIMES(
+          "Cobalt.Media.SbPlayer.Destroy.Timing", action_duration,
+          base::TimeDelta::FromMicroseconds(100),
+          base::TimeDelta::FromMilliseconds(100000), 50);
```
Comment (joeltine) [inline cobalt/media/base/metrics_provider.cc:153]: These are possibly more easily read as `FromSeconds(100)`? Here and above.

File: cobalt/media/base/metrics_provider.cc [hunk 2]
```diff
@@ -121,21 +121,38 @@ void MediaMetricsProvider::EndTrackingAction(WebMediaPlayerAction action) {
...
+    case MediaAction::SBPLAYER_CREATE_URL_PLAYER:
+      UMA_HISTOGRAM_CUSTOM_MICROSECONDS_TIMES(
+          "Cobalt.Media.SbPlayer.CreateUrlPlayer.Timing", action_duration,
+          base::TimeDelta::FromMicroseconds(100),
+          base::TimeDelta::FromMilliseconds(5000), 50);
+      break;
+    case MediaAction::SBPLAYER_DESTROY:
+      UMA_HISTOGRAM_CUSTOM_MICROSECONDS_TIMES(
+          "Cobalt.Media.SbPlayer.Destroy.Timing", action_duration,
+          base::TimeDelta::FromMicroseconds(100),
```
Comment (joeltine) [inline cobalt/media/base/metrics_provider.cc:152]: So 100μs to 100000000μs is an enormous range to be divided into 50 buckets. Do you need such a wide range? As-is, your buckets have the potential to be quite coarse and possibly not useful. I'd narrow this as much as you can. Same with the above.

Comment (sideb0ard) [reply cobalt/media/base/metrics_provider.cc:152]: woof! my mistake, they should all be in microseconds. Now corrected.

Comment (sideb0ard) [reply cobalt/media/base/metrics_provider.cc]: Adjusted range. On Linux i was seeing ~500μs for Create and ~15000μs for destroy. 
I've adjusted range to 100 - 1500μs for create, and 500 - 40000μs for destroy.

File: cobalt/media/base/metrics_provider.h [hunk 3]
```diff
@@ -78,20 +81,20 @@ class MediaMetricsProvider {
...
 
-  void ReportActionLatencyUMA(WebMediaPlayerAction action,
+  void ReportActionLatencyUMA(MediaAction action,
                               const base::TimeDelta& action_duration);
 
  private:
   // Media player action latency data.
   const base::TickClock* clock_;
-  base::small_map<std::map<WebMediaPlayerAction, base::TimeTicks>, 2>
+  base::small_map<std::map<MediaAction, base::TimeTicks>, 2>
```
Comment (at-ninja) [inline cobalt/media/base/metrics_provider.h:97]: I think this should be increased to 5.

Comment (sideb0ard) [reply cobalt/media/base/metrics_provider.h]: updated, thanks!

**Review by joeltine — APPROVED** [review]

**Comment by kaidokert** [comment]
This broke internal builds - did we run the presubmits for those ?

**Comment by sideb0ard** [comment]
> This broke internal builds - did we run the presubmits for those ?

yeah, pre-submits passed. This was a Mac only issue, which aren't run for presubmits though. Was a missing underscore - https://github.com/youtube/cobalt/pull/2851

**Comment by kaidokert** [comment]
> This was a Mac only issue, which aren't run for presubmits though

They are, you have to trigger them internally.

**Pull request merged**